runtime/src/iree/hal/drivers/hip/hip_device.c

```diff
@@ -30,10 +30,12 @@
 #include "iree/hal/drivers/hip/rccl_dynamic_symbols.h"
 #include "iree/hal/drivers/hip/status_util.h"
 #include "iree/hal/drivers/hip/stream_command_buffer.h"
+#include "iree/hal/utils/caching_allocator.h"
```

Comment:
Allocators are injected and cannot be depended on here - this creates a strong dependency on the allocator library and we cannot have that. There should not be a need to know what the allocator is providing the buffers, and there may be multiple nested allocators (you can wrap a debug allocator around a caching allocator and then around a base allocator).